src/core/column/reduce_unary.h

```diff
@@ -26,15 +26,16 @@
 namespace dt {
 
 
-template <typename T, bool IS_GROUPED>
+template <typename T, typename U, bool IS_GROUPED>
```

Comment:
I guess since we now have two types here, it would be nice to have either some intuitive names like `T_IN` and `T_OUT`, or add some comments as to what `T` and `U` mean. The former is probably more preferable.